In [16]:
!head /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3 -n 10

##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Myoluc2.0
#!genome-build-accession NCBI_Assembly:GCF_000147115.1
#!annotation-date 26 January 2018
#!annotation-source NCBI Myotis lucifugus Annotation Release 102
##sequence-region NW_005871048.1 1 64990205
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=59463
NW_005871048.1	RefSeq	region	1	64990205	.	+	.	ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female


In [ ]:
GL429891 266596

In [6]:
!wc -l /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf

482554 /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf


In [7]:
!tail /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf

AAPE02072733	ensembl	gene	1916	3700	.	+	.	gene_id "ENSMLUG00000027989"; gene_version "1"; gene_source "ensembl"; gene_biotype "protein_coding";
AAPE02072733	ensembl	transcript	1916	3700	.	+	.	gene_id "ENSMLUG00000027989"; gene_version "1"; transcript_id "ENSMLUT00000024269"; transcript_version "1"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_source "ensembl"; transcript_biotype "protein_coding";
AAPE02072733	ensembl	exon	1916	2309	.	+	.	gene_id "ENSMLUG00000027989"; gene_version "1"; transcript_id "ENSMLUT00000024269"; transcript_version "1"; exon_number "1"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_source "ensembl"; transcript_biotype "protein_coding"; exon_id "ENSMLUE00000256822"; exon_version "1";
AAPE02072733	ensembl	CDS	1916	2309	.	+	0	gene_id "ENSMLUG00000027989"; gene_version "1"; transcript_id "ENSMLUT00000024269"; transcript_version "1"; exon_number "1"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_source "ensemb

In [2]:
!head '/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNYVT40_margCompvalues.csv' -n 44

"","SNP","Scaffold","Maj","Min","NJ","NY","VT","Comp"
"15",282162,"GL429849","A","G",0.00111399888600111,0.00836899163100837,9.99999000001e-07,3.25319264647375e-09
"19",453486,"GL429872","A","G",0.000886999113000887,9.99999000001e-07,0.00143799856200144,5.14576123265913e-10
"20",463033,"GL429852","G","A",7.4999925000075e-05,0.000112999887000113,0.000434999565000435,1.37829106611689e-09
"21",463041,"GL429852","G","A",7.6999923000077e-05,1.999998000002e-06,0.00203599796400204,1.39290583823221e-10
"26",616121,"GL429834","G","C",9.99999000001e-07,0.000861999138000862,8.1999918000082e-05,3.46164465967714e-11
"51",1543734,"GL429882","T","C",0.00874799125200875,9.99999000001e-07,9.99999000001e-07,4.87504391237632e-12
"55",1735029,"GL429966","C","T",0.00301799698200302,0.00946499053500947,9.99999000001e-07,9.14116512034742e-09
"69",3296417,"GL429813","C","T",0.00786199213800786,9.99999000001e-07,0.00244999755000245,6.35734734029988e-09
"77",6378055,"GL429769","A","G",9.99999000001e-07,0.007934

In [21]:
!head /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf

#!genome-build Myoluc2.0
#!genome-version Myoluc2.0
#!genome-date 2010-09
#!genome-build-accession NCBI:GCA_000147115.1
#!genebuild-last-updated 2011-06
GL429767	ensembl	gene	42604	226589	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding";
GL429767	ensembl	transcript	42604	226589	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; transcript_id "ENSMLUT00000025902"; transcript_version "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ACHE-201"; transcript_source "ensembl"; transcript_biotype "protein_coding";
GL429767	ensembl	exon	42604	43700	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; transcript_id "ENSMLUT00000025902"; transcript_version "1"; exon_number "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ACHE-201"; transcript_source "ensembl"; transcript_biotype "protein_coding"; exon_id "ENSMLUE00000285193"; exon

In [22]:
!head -n 92 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","NY","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000181999818000182,2.90413319116625e-07,70,5.39909603706351e-07
"2","GL429767","NW_005871048.1",38456248,"T","C",2.8999971000029e-05,0.000823999176000824,4.43259321723044e-07,78,6.01613558415648e-07
"3","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,9.99999000001e-07,1.45904282432881e-08,20,1.54259886773243e-07
"4","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,1.63656087014458e-09,9,6.94169490479594e-08
"5","GL429767","NW_005871048.1",51542832,"G","A",9.99999000001e-07,0.000233999766000234,5.42310342192109e-09,13,1.00268926402608e-07
"6","GL429768","NW_005871049.1",3921142,"G","C",9.99999000001e-07,0.0138399861600138,2.64284027064383e-07,67,5.16770620690364e-07
"7","GL429768","NW_005871049.1",34577793,"C","T",0.000584999415000585,2.3999976000024e-05,2.6790172263069e-07,68,5.2448361502902

In [ ]:
2,125,561-2,462,498   2,603,829

In [ ]:
46,836,518..47,675,316

In [ ]:
2,123,766, 3,029,105

MASP1- A
GALNT18- A
BMPER
GNPDA1
BFSP1
REPS2 (REPS1 A)
PCDH17 ( PCDH15 A PCDH11)
SOX5-A
MDGA2-A
CACNA1D
FA2H
TPRA1
ADCY3- A
CEP112
TG -A 
NELL1 - A
SEMA5A -A
SORCS3 - A
CBFA2T3 SHOULD 
SGCZ -A 
GEMIN4 (GEMIN7 A)
MKL2 
BACE2
RAPGEF1 -A 

There are 1,591 genes in that set...can we see how many are in the genome, and see how many genes are in our genome. 

can we do something like epstein We sampled 1,000 pairs of 200 kb regions around randomly selected SNPs in our data set, and only 25% of such regions contained one or more genes with Gene Ontology14 terms ‘immune response,’ ‘cell proliferation,’ or ‘cell death’ (our candidate regions contained both an immune response and a cell death GO annotation). Consequently, our result (two out of seven based on GO terms or five out of seven based on manual annotation) is unlikely to have been found by chance.




MASP1
GALNT18
BMPER
GNPDA1
BFSP1
REPS2
PCDH17
SOX5
MDGA2
CACNA1D- GNRH, INSULIN
FA2H
TPRA1
ADCY3- GNRH, INSULIN
CEP112
TG
NELL1
SEMA5A
SORCS3
CBFA2T3
SGCZ
GEMIN4
MKL2
BACE2
RAPGEF1- RAPGEF4 INSULIN


In [15]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout2_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    if args[2] in chrom_dict:
        chrom_dict[args[2]].append(int(args[3]))
    else:
        chrom_dict[args[2]] = [int(args[3])]
        
aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        for loc in chrom_dict[chrom]:
        # loc = chrom_dict[chrom]
            if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
                count += 1
                print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chrom

['NW_005871087.1', 'RefSeq', 'region', '1', '8424078', '.', '+', '.', 'ID=id142776;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871097.1', 'RefSeq', 'region', '1', '7348857', '.', '+', '.', 'ID=id162626;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871097.1', 'Gnomon', 'gene', '3142067', '3321329', '.', '-', '.', 'ID=gene8121;Dbxref=GeneID:102440988;Name=REPS2;gbkey=Gene;gene=REPS2;gene_biotype=protein_coding\n']
['NW_005871097.1', 'Gnomon', 'mRNA', '3142067', '3321329', '.', '-', '.', 'ID=rna13665;Parent=gene8121;Dbxref=GeneID:102440988,Genbank:XM_023753740.1;Name=XM_023753740.1;gbkey=mRNA;gene=REPS2;model_evidence=Supporting evidence includes similarity to: 1 Protein%2C and 82%25 coverage of the annotated genomic feature by RNAseq alignments;product=RALBP1 associated Eps domain 

['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_006090529.3;Name=XM_006090529.3;gbkey=mRNA;gene=PCDH17;model_evidence=Supporting evidence includes similarity to: 2 mRNAs%2C 5 Proteins%2C and 95%25 coverage of the annotated genomic feature by RNAseq alignments;product=protocadheri

['NW_005871219.1', 'RefSeq', 'region', '1', '2865968', '.', '+', '.', 'ID=id315315;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871228.1', 'RefSeq', 'region', '1', '2723584', '.', '+', '.', 'ID=id320747;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871231.1', 'RefSeq', 'region', '1', '2651257', '.', '+', '.', 'ID=id323591;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871236.1', 'RefSeq', 'region', '1', '2551290', '.', '+', '.', 'ID=id327813;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871243.1', 'RefSeq', 'region', '1', '2513712', '.', '+', '.', 'ID=id331702;Dbxref=taxon:5946

['NW_005871763.1', 'RefSeq', 'region', '1', '283313', '.', '+', '.', 'ID=id502584;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871846.1', 'RefSeq', 'region', '1', '211799', '.', '+', '.', 'ID=id512757;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:1024175

160

In [1]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout1.5_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    if args[2] in chrom_dict:
        chrom_dict[args[2]].append(int(args[3]))
    else:
        chrom_dict[args[2]] = [int(args[3])]
        
aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        for loc in chrom_dict[chrom]:
        # loc = chrom_dict[chrom]
            if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
                count += 1
                print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chrom

['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_006090529.3;Name=XM_006090529.3;gbkey=mRNA;gene=PCDH17;model_evidence=Supporting evidence includes similarity to: 2 mRNAs%2C 5 Proteins%2C and 95%25 coverage of the annotated genomic feature by RNAseq alignments;product=protocadheri

['NW_005871210.1', 'Gnomon', 'gene', '2499532', '3006516', '.', '+', '.', 'ID=gene15442;Dbxref=GeneID:102425362;Name=SORCS3;gbkey=Gene;gene=SORCS3;gene_biotype=protein_coding\n']
['NW_005871210.1', 'Gnomon', 'mRNA', '2499532', '3006516', '.', '+', '.', 'ID=rna26668;Parent=gene15442;Dbxref=GeneID:102425362,Genbank:XM_006095936.3;Name=XM_006095936.3;gbkey=mRNA;gene=SORCS3;model_evidence=Supporting evidence includes similarity to: 4 mRNAs%2C 11 Proteins%2C and 30%25 coverage of the annotated genomic feature by RNAseq alignments;product=sortilin related VPS10 domain containing receptor 3;transcript_id=XM_006095936.3\n']
['NW_005871219.1', 'RefSeq', 'region', '1', '2865968', '.', '+', '.', 'ID=id315315;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871228.1', 'RefSeq', 'region', '1', '2723584', '.', '+', '.', 'ID=id320747;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Fram

['NW_005871763.1', 'RefSeq', 'region', '1', '283313', '.', '+', '.', 'ID=id502584;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:102417583;Name=RAPGEF1;gbkey=Gene;gene=RAPGEF1;gene_biotype=protein_coding\n']
['NW_005872028.1', 'Gnomon', 'mRNA', '5481', '98250', '.', '-', '.', 'ID=rna46592;Parent=gene27256;Dbxref=GeneID:102417583,Genbank:XM_023747795.1;Name=X

140

In [3]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout1.5_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    chrom_dict[args[2]] = int(args[3])

aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        loc = chrom_dict[chrom]
        if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
            count += 1
            print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871049.1', 'RefSeq', 'region', '1', '39494369', '.', '+', '.', 'ID=id14505;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871051.1', 'RefSeq', 'region', '1', '30916457', '.', '+', '.', 'ID=id28999;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'RefSeq', 'region', '1', '20945196', '.', '+', '.', 'ID=id44306;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'Gnomon', 'gene', '8711696', '8718521', '.', '+', '.', 'ID=gene2288;Dbxref=GeneID:111

['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_006090529.3;Name=XM_006090529.3;gbkey=mRNA;gene=PCDH17;model_evidence=Supporting evidence includes similarity to: 2 mRNAs%2C 5 Proteins%2C and 95%25 coverage of the annotated genomic feature by RNAseq alignments;product=protocadherin 17;transcript_id=XM_006090529.3\n']
['NW_005871123.1', 'RefSeq', 'region', '1', '5160323', '.', '+', '.', 'ID=id201293;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomi

['NW_005871219.1', 'RefSeq', 'region', '1', '2865968', '.', '+', '.', 'ID=id315315;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871228.1', 'RefSeq', 'region', '1', '2723584', '.', '+', '.', 'ID=id320747;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871236.1', 'RefSeq', 'region', '1', '2551290', '.', '+', '.', 'ID=id327813;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871243.1', 'RefSeq', 'region', '1', '2513712', '.', '+', '.', 'ID=id331702;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871246.1', 'RefSeq', 'region', '1', '2532579', '.', '+', '.', 'ID=id335396;Dbxref=taxon:5946

['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:102417583;Name=RAPGEF1;gbkey=Gene;gene=RAPGEF1;gene_biotype=protein_coding\n']
['NW_005872028.1', 'Gnomon', 'mRNA', '5481', '98250', '.', '-', '.', 'ID=rna46592;Parent=gene27256;Dbxref=GeneID:102417583,Genbank:XM_023747795.1;Name=XM_023747795.1;gbkey=mRNA;gene=RAPGEF1;model_evidence=Supporting evidence includes similarity to: 7 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 1 sample with support for 

115

In [9]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    chrom_dict[args[2]] = int(args[3])

aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        loc = chrom_dict[chrom]
        if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
            count += 1
            print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871049.1', 'RefSeq', 'region', '1', '39494369', '.', '+', '.', 'ID=id14505;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871051.1', 'RefSeq', 'region', '1', '30916457', '.', '+', '.', 'ID=id28999;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'RefSeq', 'region', '1', '20945196', '.', '+', '.', 'ID=id44306;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'Gnomon', 'gene', '14979316', '15072600', '.', '-', '.', 'ID=gene2361;Dbxref=GeneID:1

['NW_005871109.1', 'RefSeq', 'region', '1', '6495742', '.', '+', '.', 'ID=id181624;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871109.1', 'Gnomon', 'gene', '2888188', '3176295', '.', '+', '.', 'ID=gene9224;Dbxref=GeneID:102435828;Name=GALNT18;gbkey=Gene;gene=GALNT18;gene_biotype=protein_coding\n']
['NW_005871109.1', 'Gnomon', 'mRNA', '2888188', '3176295', '.', '+', '.', 'ID=rna15475;Parent=gene9224;Dbxref=GeneID:102435828,Genbank:XM_006089610.3;Name=XM_006089610.3;gbkey=mRNA;gene=GALNT18;model_evidence=Supporting evidence includes similarity to: 6 mRNAs%2C 7 Proteins%2C and 99%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 9 samples with support for all annotated introns;product=polypeptide N-acetylgalactosaminyltransferase 18%2C transcript variant X1;transcript_id=XM_006089610.3\n']
['NW_005871109.1', 'Gnomon', 'mRNA', '2888188', '3176292', '.', '+'

['NW_005871148.1', 'RefSeq', 'region', '1', '4444721', '.', '+', '.', 'ID=id238203;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871148.1', 'Gnomon', 'gene', '2560994', '2842740', '.', '-', '.', 'ID=gene11898;Dbxref=GeneID:102438113;Name=CEP112;gbkey=Gene;gene=CEP112;gene_biotype=protein_coding\n']
['NW_005871148.1', 'Gnomon', 'mRNA', '2560994', '2842740', '.', '-', '.', 'ID=rna20342;Parent=gene11898;Dbxref=GeneID:102438113,Genbank:XM_014458887.2;Name=XM_014458887.2;gbkey=mRNA;gene=CEP112;model_evidence=Supporting evidence includes similarity to: 2 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments;product=centrosomal protein 112;transcript_id=XM_014458887.2\n']
['NW_005871154.1', 'RefSeq', 'region', '1', '4280553', '.', '+', '.', 'ID=id250845;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;m

['NW_005871263.1', 'RefSeq', 'region', '1', '2259905', '.', '+', '.', 'ID=id349927;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871279.1', 'RefSeq', 'region', '1', '1996049', '.', '+', '.', 'ID=id357671;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871289.1', 'RefSeq', 'region', '1', '1901218', '.', '+', '.', 'ID=id361809;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871299.1', 'RefSeq', 'region', '1', '1811804', '.', '+', '.', 'ID=id366663;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871310.1', 'RefSeq', 'region', '1', '1841132', '.', '+', '.', 'ID=id371601;Dbxref=taxon:5946

['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871977.1', 'RefSeq', 'region', '1', '146487', '.', '+', '.', 'ID=id520514;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:102417583;Name=RAPGEF1;gbkey=Gene;gene=RAPGEF1;gene_biotype=protein_coding\n']
['NW_005872028.1', 'Gnomon', 'mRNA', '5481', '98250', '.', '-', '.', 'ID=rna46592;Parent=gene27256;Dbxref=GeneID:102417583,Genbank:XM_023747795.1;Name=X

145

In [ ]:
p vals 

2603829 MDGA2,  in our sig all 3
616121 unknown, but in our sig all 3
35761989 unknown, but in our sig all 3

## VEP for synonymous/non synonymous

create input file of sig snps

The default format is a simple whitespace-separated format (columns may be separated by space or tab characters), containing five required columns plus an optional identifier column:

chromosome - just the name or number, with no 'chr' prefix
start
end
allele - pair of alleles separated by a '/', with the reference allele first
strand - defined as + (forward) or - (reverse).
identifier - this identifier will be used in VEP's output. If not provided, VEP will construct an identifier from the given coordinates and alleles.

In [18]:
!head /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","NY","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000181999818000182,2.90413319116625e-07,70,5.39909603706351e-07
"2","GL429767","NW_005871048.1",38456248,"T","C",2.8999971000029e-05,0.000823999176000824,4.43259321723044e-07,78,6.01613558415648e-07
"3","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,9.99999000001e-07,1.45904282432881e-08,20,1.54259886773243e-07
"4","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,1.63656087014458e-09,9,6.94169490479594e-08
"5","GL429767","NW_005871048.1",51542832,"G","A",9.99999000001e-07,0.000233999766000234,5.42310342192109e-09,13,1.00268926402608e-07
"6","GL429768","NW_005871049.1",3921142,"G","C",9.99999000001e-07,0.0138399861600138,2.64284027064383e-07,67,5.16770620690364e-07
"7","GL429768","NW_005871049.1",34577793,"C","T",0.000584999415000585,2.3999976000024e-05,2.6790172263069e-07,68,5.2448361502902

In [26]:
!awk 'BEGIN { FS = "," }{print $2,$4,$4,$5"/"$6, "+"}' /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv|sed 's/\"//g'| tail -n 92 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP.txt    

In [28]:
!awk 'BEGIN { FS = "," }{print $3,$4,$4,$5"/"$6, "+"}' /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv|sed 's/\"//g'| tail -n 92 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt    

In [ ]:
/local/home/sarahgw/ensembl-vep/vep -i /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP.txt --cache --species myotis_lucifugus 

bgzip and tabix file

In [23]:
!head /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3

##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Myoluc2.0
#!genome-build-accession NCBI_Assembly:GCF_000147115.1
#!annotation-date 26 January 2018
#!annotation-source NCBI Myotis lucifugus Annotation Release 102
##sequence-region NW_005871048.1 1 64990205
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=59463
NW_005871048.1	RefSeq	region	1	64990205	.	+	.	ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female


In [29]:
!grep -v "#" /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3 |  sort -k1,1 -k4,4n > /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3

In [ ]:
bgzip -c /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3 > /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz
tabix -p gff /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz

In [ ]:
bgzip -c /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa > /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa.gz

In [30]:
/local/home/sarahgw/ensembl-vep/vep  -i /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt -fasta /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa --gff /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz --cache --species myotis_lucifugus 


SyntaxError: invalid syntax (<ipython-input-30-f0842e4a7489>, line 1)

In [1]:
!head -n 100 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt

NW_005871048.1 16839644 16839644 T/A +
NW_005871048.1 38456248 38456248 T/C +
NW_005871048.1 22649683 22649683 T/A +
NW_005871048.1 46534125 46534125 C/T +
NW_005871048.1 51542832 51542832 G/A +
NW_005871049.1 3921142 3921142 G/C +
NW_005871049.1 34577793 34577793 C/T +
NW_005871049.1 29962249 29962249 A/G +
NW_005871051.1 8292792 8292792 A/T +
NW_005871051.1 8507307 8507307 A/C +
NW_005871053.1 8716095 8716095 T/A +
NW_005871053.1 14993622 14993622 T/C +
NW_005871054.1 7204887 7204887 C/T +
NW_005871056.1 5556309 5556309 A/C +
NW_005871059.1 3561000 3561000 G/A +
NW_005871062.1 7594948 7594948 A/T +
NW_005871064.1 14617697 14617697 C/A +
NW_005871064.1 665407 665407 C/T +
NW_005871065.1 1229639 1229639 G/A +
NW_005871066.1 3216149 3216149 C/T +
NW_005871073.1 753145 753145 C/A +
NW_005871075.1 5707870 5707870 T/A +
NW_005871076.1 3907289 3907289 G/A +
NW_005871082.1 6986210 6986210 A/T +
NW_005871090.1 7390034 7390034 A/G +
NW_005871097.1 3229647 3229647 G/A +
NW_005871100.1 365354 36

In [5]:
!tail -n 80 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/outputVEP2

NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_006090548.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_006090549.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_014457399.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871126.1_2091046_C/T	NW_005871126.1:2091046	T	102433978	XM_006090652.3	Transcript	intron_variant	-	-	-	-	-	-	IMPACT=MODIFIER;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871129.1_2603829_T/C	NW_005871129.1:2603829	C	102423892	XM_006090751.3	Transcript	intron_variant	-	-	-	-	-	-	IMPACT=MODIFIER;STRAND=-1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
